submissions/antonkottans/memory-pair-game/script.js

```diff
@@ -0,0 +1,250 @@
+    });
+};
+
+const getCardName = (node) => {
+    return node.getAttribute("src").slice(9, -4) || undefined;
+};
+
+const clickHandler = ({ target }) => {
+    if (
+        target.matches(".back") &&
+        target.parentNode.childElementCount === 2 &&
+        !target.closest('*[class*="card_"].flip-to-face')
+    ) {
+        target.closest('*[class*="card_"]').classList.add("flip-to-face");
+
+        setTimeout(() => {
+            const currentCardContainer = target.closest('*[class*="card_"]');
+            const currentCardFace =
+                target.closest('*[class*="card_"]').children[1];
+
+            if (!state.flipped) {
+                state.flipped = {
+                    container: currentCardContainer,
+                    face: currentCardFace,
+                };
+            } else {
+                if (
+                    getCardName(state.flipped.face) ===
+                    getCardName(currentCardFace)
+                ) {
+                    currentCardContainer.classList.remove("flip-to-face");
+                    state.flipped.container.classList.remove("flip-to-face");
+                    state.guessedIds.add(state.flipped.face.getAttribute("id"));
+                    state.guessedIds.add(currentCardFace.getAttribute("id"));
+                    state.flipped = false;
+                    state.guessedPairsAmount++;
+                    if (state.guessedPairsAmount === 6) {
+                        state.guessedPairsAmount = 0;
+                        stopGame("win");
+                    }
+                } else {
+                    currentCardContainer.classList.remove("flip-to-face");
+                    state.flipped.container.classList.remove("flip-to-face");
+
+                    state.flipped = false;
+                }
+            }
+            antiCheat();
+        }, 1000);
+    }
+};
+
+const startGame = () => {
+    clearInterval(timerId);
+    state.guessedPairs = 0;
+    state.guessedIds = new Set();
+    document
+        .querySelectorAll(".guessed")
+        .forEach((elem) => elem.classList.remove("guessed"));
+    setTimeout(() => {
+        const cardImageNodes = getDoubledRandomImageNodes(6);
+        state.notGuessedIds = new Set(
+            cardImageNodes.map((node) => node.getAttribute("id"))
+        );
+        moveCardsIntoDOM(cardImageNodes);
+        startTimer();
+    }, 1000);
+};
+
+const stopGame = (result = "win") => {
+    clearInterval(timerId);
+    const span = document.createElement("span");
+    span.classList.add("result");
+    span.textContent =
+        result === "win"
+            ? `You won! You time:${state.timerNode.textContent}`
+            : "You lost!";
+    state.bodyNode.appendChild(span);
+    span.addEventListener("click", () => {
+        state.bodyNode.removeChild(span);
+    });
+};
+
+const getDoubledRandomImageNodes = (amountOfElements) => {
+    if (amountOfElements <= 0) return [];
+
+    const getCardNames = (amount) => {
```

Comment:
This function doesn't return Card Names. It's misleading. It returns another function that, if called, returns card names. I believe you've made it that complicated because of pipe. There is no reason for it.